server/schema/resolvers.js

```diff
@@ -0,0 +1,65 @@
+const { AuthenticationError } = require('apollo-server-express');
+const { User, Supplier } = require('../models');
+const { signToken } = require('../utils/auth');
+
+const resolvers = {
+  Query: {
+    users: async () => {
+      return User.find();
+    },
+    suppliers: async () => {
+        return Supplier.find();
+    },
+    supplier: async (parent, { supplierId }) => {
+      return Supplier.findOne({ _id: supplierId });
+    },
+    me: async (parent, args, context) => {
+      if (context.user) {
+        return User.findOne({ _id: context.user._id });
+      }
+      throw new AuthenticationError('You need to be logged in!');
+    },
+  },
+
+  Mutation: {
+    addUser: async (parent, { username, email, password }) => {
+      const user = await User.create({ username, email, password });
+      const token = signToken(user);
+      return { token, user };
+    },
+    login: async (parent, { email, password }) => {
+      const user = await User.findOne({ email });
+
+      if (!user) {
+        throw new AuthenticationError('No user found with this email address');
+      }
+
+      const correctPw = await user.isCorrectPassword(password);
+
+      if (!correctPw) {
+        throw new AuthenticationError('Incorrect credentials');
+      }
+
+      const token = signToken(user);
+
+      return { token, user };
+    },
+    addSupplier: async (parent, args, context) => {
+        
+    },
+    updateSupplier: async (parent, args, context) => {
+      
+    },
+    deleteSupplier: async (parent, { supplierId }, context) => {
```

Comment:
I think this argument name is incorrect, should be like this?

```suggestion
    deleteSupplier: async (parent, { _id }, context) => {
```